### PR TITLE
DRT - promised pickup time window offer acceptor

### PR DESCRIPTION
**Pull request by nkuehnel**
update of https://github.com/matsim-org/matsim-libs/pull/1954
in response to
https://github.com/matsim-org/matsim-libs/pull/1963

Left out the config parameter for now.
@michalmac I wasn't sure about the package location, it's not necessarily part of an extension.

### CODE REVIEW

**Comment by michalmac** [comment]
Nice. LGTM

>Left out the config parameter for now.

Makes sense. If we see more use cases, we can add it to the standard config. Maybe we should even group all the time constraint params into one param set. For the time being, you could try out the new possibility of subclassing the drt config.

> @michalmac I wasn't sure about the package location, it's not necessarily part of an extension.

👍

File: contribs/drt/src/main/java/org/matsim/contrib/drt/passenger/PromisedPickupTimeWindowDrtOfferAcceptor.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+	public PromisedPickupTimeWindowDrtOfferAcceptor(double promisedPickTimeWindow) {
+		this.promisedPickupTimeWindow = promisedPickTimeWindow;
+	}
+
+	@Override
+	public Optional<AcceptedDrtRequest> acceptDrtOffer(DrtRequest request, double departureTime, double arrivalTime) {
+		double updatedPickupTimeWindow = Math.min(departureTime
+				+ promisedPickupTimeWindow, request.getLatestStartTime());
+		return Optional.of(AcceptedDrtRequest.newBuilder().latestStartTime(updatedPickupTimeWindow).build());
```
Comment (michalmac) [inline contribs/drt/src/main/java/org/matsim/contrib/drt/passenger/PromisedPickupTimeWindowDrtOfferAcceptor.java:19]: For the accepted request to become fully initialised, you need to add:
```java
				.request(request)
				.earliestStartTime(request.getEarliestStartTime())
				.latestArrivalTime(request.getLatestArrivalTime())
```

Comment (nkuehnel) [reply contribs/drt/src/main/java/org/matsim/contrib/drt/passenger/PromisedPickupTimeWindowDrtOfferAcceptor.java:19]: Ah yes, forgot about that, thanks!

**Comment by nkuehnel** [comment]
> Maybe we should even group all the time constraint params into one param set

👍

**Pull request merged**